File: 3_public_encaptulaiton.py
class A:
    __value = 5
    def __init__(self) -> None:
        pass

    def display(self):
        print(self.__value)


obj = A()

# it will create a new attribute named "__value", not using the existing "__value" attribute
obj.__value = 10

print(obj.__value)
obj.display()

# print(obj.__dict__)

File: Week_0/3_public_encaptulaiton.cpp
#include <bits/stdc++.h>
using namespace std;

class A {
private:
    int value = 5;  // private attribute

public:
    // We can use, setter method to modify the private member
    void setValue(int newValue) {
        value = newValue;
    }
    
    // Crating a method to display the private member value
    // it will allow to access the value publicly
    void display() {
        cout << value << endl;
    }
};


int main()
{
    A obj;

    // setting & modifying the private member using the setter method
    obj.setValue(10);
    
    // Attempting to directly modify the private member will cause an error in C++
    // However, we can access the private value by using display functions
    obj.display();
    
    return 0;
}
